File: internal/models/product.go
package models

import "time"

type Product struct {
	ID          string         `json:"id"`
	CategoryID  string         `json:"category_id"`
	Category    *Category      `json:"category,omitempty"`
	Name        string         `json:"name"`
	Slug        string         `json:"slug"`
	Description string         `json:"description"`
	Price       float64        `json:"price"`
	Stock       int            `json:"stock"`
	ImageURL    string         `json:"image_url"`
	Images      []ProductImage `json:"images,omitempty"`
	IsActive    bool           `json:"is_active"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
}

type ProductImage struct {
	ID        string    `json:"id"`
	ProductID string    `json:"product_id"`
	ImageURL  string    `json:"image_url"`
	SortOrder int       `json:"sort_order"`
	IsPrimary bool      `json:"is_primary"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: frontend/src/types/product.ts
export type SortOrder = "asc" | "desc";

export interface Category {
  id: string;
  name: string;
  slug: string;
  description?: string | null;
  image_url?: string | null;
  parent_id?: string | null;
  parent?: Category | null;
  children?: Category[];
  created_at?: string;
  updated_at?: string;
}

export interface ProductImage {
  id: string;
  product_id: string;
  image_url: string;
  sort_order: number;
  created_at?: string;
  updated_at?: string;
}

export interface Product {
  id: string;
  name: string;
  slug: string;
  description?: string | null;
  price: number;
  stock: number;
  category_id?: string | null;
  category?: Category | null;
  image_url?: string | null;
  images?: ProductImage[];
  created_at?: string;
  updated_at?: string;
}

export interface ProductListParams {
  category_id?: string;
  category?: string;
  search?: string;
  sort_by?: string;
  sort_order?: SortOrder;
  page?: number;
  limit?: number;
}

export interface ProductListMeta {
  page: number;
  limit: number;
  total: number;
  total_pages: number;
  sort_by?: string;
  sort_order?: SortOrder | "";
  category_id?: string;
  category?: string;
  search?: string;
}

export interface ProductListResponse {
  data: Product[];
  meta: ProductListMeta;
}